playwright/_impl/_impl_to_api_mapping.py

```diff
@@ -77,14 +79,14 @@ def to_impl(self, obj: Any) -> Any:
             raise Error("Maximum argument depth exceeded")
 
     def wrap_handler(self, handler: Callable[..., None]) -> Callable[..., None]:
-        if handler in self._instances:
-            return self._instances[handler]
-
-        def wrapper(*args: Any) -> Any:
+        def wrapper_func(*args: Any) -> Any:
             arg_count = len(inspect.signature(handler).parameters)
             return handler(
                 *list(map(lambda a: self.from_maybe_impl(a), args))[:arg_count]
             )
 
-        self._instances[handler] = wrapper
+        wrapper = getattr(handler, INSTANCE_ATTR, None)
```

Comment:
It seems like this should `__pw_impl_instance`, instead of `__pw_api_instance`?

Reply:
will follow up